src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -253,6 +287,39 @@ public CheckResult Check(ParseResult parse, RecordType parameterType = null)
 
             CheckWorker(check);
             return check;
+        }
+
+        /// <summary>
+        /// Parse and resolve a type expression. 
+        /// </summary>
+        /// <param name="typeExpression">type expression in plain text. </param>
+        /// <param name="symbols">Symbols that are required to resolve the type.</param>
+        /// <param name="options">parser options to use.</param>
+        /// <returns></returns>
+        public FormulaType ResolveType(string typeExpression, SymbolTable symbols, ParserOptions options = null) 

```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

can these be static methods? Are they using anything on the Engine?

Reply:
@MikeStall since PrimitiveTypes are part of Engine and i have used that in resolvetypes and keep this non static. is that okay

Reply:
is primitiveTypes just part of incoming symbols?